libraries/standard/mqtt/src/iot_mqtt_validate.c

```diff
@@ -600,7 +600,8 @@ bool _IotMqtt_ValidateConnect( const IotMqttConnectInfo_t * pConnectInfo )
         /* If will info is provided, check that it is valid. */
         if( pConnectInfo->pWillInfo != NULL )
```

Comment:
Why are we not updating the status when pWillInfo == NULL?

Reply:
We do not update the status if `pWillInfo == NULL`, because it’s valid to have a ConnectInfo with `pWillInfo == NULL`; however, if pWillInfo is nonnull, we must check whether it's valid.

Reply:
Ok so you are saying that if the LWT is not being used it is allowed to be null and we only need to validate it if supplied. That makes sense thanks!